From: Hannes Reinecke <hare@suse.de>
Subject: Update XEN scsifront driver to request timeouts
Patch-mainline: obsolete

Update XEN scsifront driver to the block request handling
patches.

Signed-off-by: Hannes Reinecke <hare@suse.de>

Index: head-2008-09-15/drivers/xen/scsifront/scsifront.c
===================================================================
--- head-2008-09-15.orig/drivers/xen/scsifront/scsifront.c	2008-09-15 15:22:09.000000000 +0200
+++ head-2008-09-15/drivers/xen/scsifront/scsifront.c	2008-09-15 15:22:12.000000000 +0200
@@ -348,7 +348,7 @@ static int scsifront_queuecommand(struct
 		memset(ring_req->cmnd, 0, VSCSIIF_MAX_COMMAND_SIZE);
 
 	ring_req->sc_data_direction   = (uint8_t)sc->sc_data_direction;
-	ring_req->timeout_per_command = (sc->timeout_per_command / HZ);
+	ring_req->timeout_per_command = (sc->request->timeout / HZ);
 
 	info->shadow[rqid].req_scsi_cmnd     = (unsigned long)sc;
 	info->shadow[rqid].sc_data_direction = sc->sc_data_direction;
@@ -418,7 +418,7 @@ static int scsifront_dev_reset_handler(s
 		memset(ring_req->cmnd, 0, VSCSIIF_MAX_COMMAND_SIZE);
 
 	ring_req->sc_data_direction   = (uint8_t)sc->sc_data_direction;
-	ring_req->timeout_per_command = (sc->timeout_per_command / HZ);
+	ring_req->timeout_per_command = (sc->request->timeout / HZ);
 	ring_req->nr_segments         = 0;
 
 	scsifront_do_request(info);	
